packages/api/src/pins.js

```diff
@@ -261,85 +263,176 @@ export async function pinGet (request, env, ctx) {
-      { error: { reason: ERROR_STATUS, details: INVALID_NAME } },
-      { status: ERROR_CODE }
-    )
+  if (name) {
+    if (typeof name !== 'string') {
+      return {
+        error: { reason: ERROR_STATUS, details: INVALID_NAME },
+        data: undefined
+      }
+    }
+    opts.name = name
   }
 
-  // Validate match
-  if (match && typeof match !== 'string') {
-    if (!MATCH_OPTIONS.includes(match)) {
-      return new JSONResponse(
-        { error: { reason: ERROR_STATUS, details: UNPERMITTED_MATCH } },
-        { status: ERROR_CODE }
-      )
+  if (match) {
+    if (typeof match !== 'string') {
+      if (!MATCH_OPTIONS.includes(match)) {
+        return {
+          error: { reason: ERROR_STATUS, details: UNPERMITTED_MATCH },
+          data: undefined
+        }
+      }
+
+      return {
+        error: { reason: ERROR_STATUS, details: INVALID_MATCH },
+        data: undefined
+      }
     }
+    opts.match = match
+  }
 
-    return new JSONResponse(
-      { error: { reason: ERROR_STATUS, details: INVALID_MATCH } },
-      { status: ERROR_CODE }
-    )
+  if (status) {
+    if (Array.isArray(status)) {
+      const isValidStatus = status.every(v => STATUS_OPTIONS.includes(v))
+      if (!isValidStatus) {
+        return {
+          error: { reason: ERROR_STATUS, details: UNPERMITTED_STATUS },
+          data: undefined
+        }
+      }
+    } else {
+      return {
+        error: { reason: ERROR_STATUS, details: INVALID_STATUS },
+        data: undefined
+      }
+    }
+    opts.status = status
   }
 
-  // Validate status
-  if (status && !Array.isArray(status)) {
-    const isValidStatus = status.every(v => STATUS_OPTIONS.includes(v))
-    if (!isValidStatus) {
-      return new JSONResponse(
-        { error: { reason: ERROR_STATUS, details: UNPERMITTED_STATUS } },
-        { status: ERROR_CODE }
-      )
+  if (before) {
+    if ((typeof before !== 'string' || (new Date(before)).getTime() <= 0)) {
+      return {
+        error: { reason: ERROR_STATUS, details: INVALID_TIMESTAMP },
+        data: undefined
+      }
     }
+    opts.before = before
+  }
 
-    return new JSONResponse(
-      { error: { reason: ERROR_STATUS, details: INVALID_STATUS } },
-      { status: ERROR_CODE }
-    )
+  if (after) {
+    if ((typeof after !== 'string' || (new Date(after)).getTime() <= 0)) {
+      return {
+        error: { reason: ERROR_STATUS, details: INVALID_TIMESTAMP },
+        data: undefined
+      }
+    }
+    opts.after = after
+  }
+
+  if (limit) {
+    if (!(/^\d+$/.test(limit))) {
+      return {
+        error: { reason: ERROR_STATUS, details: INVALID_LIMIT },
+        data: undefined
+      }
+    }
+    opts.limit = limit
   }
 
-  // TODO: write logic for listing pins
-  return new JSONResponse('OK')
+  return { error: undefined, data: opts }
+}
+
+/**
+ * Transform a PinRequest into a PinStatus
+ *
+ * @param { Object } pinRequest
+ * @returns { ServiceApiPinStatus }
+ */
+function getPinStatus (pinRequest) {
```

Comment:
Reallly nice 👌 

Can we please use this in the get `pinGet` endpoint?

Reply:
I believe it already is used from `pinGet`

Reply:
@GaryHomewood, sorry, I meant pinPost, specifically in the pinCreate function.

Reply:
Ta